drivers/serial/uart_nrfx_uarte.c

```diff
@@ -524,26 +576,39 @@ static int uarte_instance_init(struct device *dev,
 	return 0;
 }
 
+
 #define UART_NRF_UARTE_DEVICE(idx)					       \
 	DEVICE_DECLARE(uart_nrfx_uarte##idx);				       \
 	UARTE_##idx##_CREATE_TX_BUFF;					       \
 	static struct uarte_nrfx_data uarte_##idx##_data = {		       \
 		UARTE_##idx##_DATA_INIT					       \
+		.uart_config = {					       \
+			.baudrate =					       \
+				DT_NORDIC_NRF_UARTE_UART_##idx##_CURRENT_SPEED,\
+			.data_bits = UART_CFG_DATA_BITS_8,		       \
+			.stop_bits = UART_CFG_STOP_BITS_1,		       \
+			.parity = UARTE_##idx##_NRF_PARITY_BIT		       \
+				  == NRF_UARTE_PARITY_INCLUDED		       \
```

Comment:
This could be indented one level more to improve readibility. Same applies to `== NRF_UARTE_HWFC_ENABLED` below.